clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -368,7 +373,11 @@ public void onSuccess(Void value) {
 
             @Override
             public void onFailure(RuntimeException e) {
-                cb.onComplete(offsets, e);
+                if (e instanceof RetriableException) {
+                    cb.onComplete(offsets, RETRIABLE_COMMIT_EXCEPTION);
```

Comment:
Depends on the user I guess. I can imagine wanting to retry a commit, but I'd probably just use commitSync() in that case. If we want to discourage commit retries (because the user will have to ensure commit ordering themselves), then we could treat all commits as fatal. However, I feel it might be better to err on the side of exposing more information so that users can decided for themselves what to do. Thoughts?